src/components/Calendars/ClientScheduleAppointment.tsx

```diff
@@ -0,0 +1,342 @@
+'use client';
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { getDaysInMonth, isToday, isSameDay } from 'date-fns';
+
+type ClientScheduleAppointmentProps = {
+    availableDates: Date[];
+};
+
+const MONTH_NAMES = [
+    'Janeiro',
+    'Fevereiro',
+    'Março',
+    'Abril',
+    'Maio',
+    'Junho',
+    'Julho',
+    'Agosto',
+    'Setembro',
+    'Outubro',
+    'Novembro',
+    'Dezembro',
+];
+
+const DAYS = ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sáb'];
+
+// Calculates the number of days in the given month
+const getNoOfDays = (date: Date) => {
+    const daysInMonth = getDaysInMonth(date);
+    const dayOfWeek = date.getDay();
+
+    const blankdaysArray: number[] = [];
+    for (var i = 1; i <= dayOfWeek; i++) {
+        blankdaysArray.push(i);
+    }
+
+    const daysArray: number[] = [];
+    for (var i = 1; i <= daysInMonth; i++) {
+        daysArray.push(i);
+    }
+
+    return { blankDays: blankdaysArray, numDays: daysArray };
+};
+
+const isSelectedDate = (newDate: Date, selectedDate: Date) => {
+    return isSameDay(newDate, selectedDate);
+};
+
+const isSelectedHour = (hour: number, selectedHour: number | undefined) => {
+    return selectedHour === hour;
+};
+
+const submitDate = (selectedDate: Date) => {
+    if (selectedDate) {
+        //send the selectedDate object
+    }
+};
+
+export default function ClientScheduleAppointment({ availableDates }: ClientScheduleAppointmentProps) {
+    const [month, setMonth] = useState(new Date().getMonth());
+    const [year, setYear] = useState(new Date().getFullYear());
+    const [selectedDate, setSelectedDate] = useState<Date>();
+    const [selectedHour, setSelectedHour] = useState<number | undefined>();
+    const [step, setStep] = useState(0);
+
+    const selectDateValue = useCallback(
+        (day: number) => {
+            const selectedDate = new Date(year, month, day);
+
+            setSelectedDate(selectedDate);
+            setSelectedHour(undefined);
+            setStep(1);
+        },
+        [month, year],
+    );
+
+    const hasNoAvailability = (day: number): boolean => {
+        return !availableDates.find((d: Date) => isSameDay(d, new Date(year, month, day)));
+    };
+
+    const DATE_SELECTION = 0;
+    const HOUR_SELECTION = 1;
```

Comment:
put these constants on the top of the file so we dont always recreate them as well